Add spec.build.env to workload

emmjohnson opened the pull request.

Commit:
Add spec.build.env to workload

cirocosta reviewed (approved):
overall lgtm! 

we should update the reference docs before merging though

Commit:
examples: bump kpack, use spec.build.env

here we add to the example the use of workload.spec.build.env to
showcase how one can go about tweaking the environment variables passed
solely to the build stage of our supplychain.

we can verify that kpack indeed does so by looking at the logs from the
`prepare` container:

	Container/prepare
		Build reason(s): CONFIG
		CONFIG:
		      + env:
		      + - name: CGO_ENABLED
		      +   value: "0"
		      resources: {}
		      - source: {}
		      + source:
		      +   blob:
		      +     url: http://source-controller.gito...
		Loading secret for "10.188.0.3:5000" from secr...
		Downloading source-controller.gitops-toolkit.s...
		Successfully downloaded source-controller.gito...

which indeed gets propagated to the pod:

    - name: PLATFORM_ENV_VARS
      value: '[{"name":"CGO_ENABLED","value":"0"}]'

Signed-off-by: Ciro S. Costa <ciroscosta@vmware.com>

Commit:
docs: update reference to include spec.build.env

Signed-off-by: Ciro S. Costa <ciroscosta@vmware.com>
